Kevinf sept updates

kfansler opened the pull request:
This is ready to go. Lots of random fixes. I’ll change the items in the Stage column of the Product Backlog sheet to PR so it’s easier to cross-reference.

Commit:
Added "impersonate" to assume-user description

To aid in searching

Commit:
Clarified that query strings are case sensitive

Commit:
Clarified queries and case-sensitivity

Commit:
Gave camelCase hint

Commit:
Closed an HTML call

Commit:
Pagination clarification

Strongly typed languages include all objects even if it’s an empty
array.

Commit:
Inserted in the wrong place

Saving for later.

Commit:
Object details vs summary info topic

Commit:
Edited for readability

Commit:
Extra hint about summary data

Commit:
New logic for Row Location

Logic worked out with Andrew Calvin, but he hasn’t seen the new draft
yet.a

Commit:
Brought Row Location into current style

Put attribute in first column, description in second.

Commit:
Incorporated review feedback

* Defined sibling and the hierarchical model
* Changed order of table rows to be more logical

Commit:
Typo fix

Commit:
Less chatty

Added a Type column to make the table scan better.
Removed hierarchical term and replaced it with “indent level”

Commit:
Changed line endings of comments

To fit new template

Commit:
Clarified Predecessor rules

There were still some sections that mentioned that you could not update
Predecessors through the API.

Commit:
Created cross-references

Instead of repeating the information in 4 places.

Commit:
New Error Object topic

Cleaned up links. Most point to the Error Object.

Commit:
Proper alpha order

This seems to have been added to the bottom of the list. I moved it to
the correct alpha order for “other objects”

Commit:
Other Object cleanup

* Example object placeholders
* Example AutoNumberFormat Object
* Example ContainerDestination Object
* Fixed object comma issue with Move Folder

Commit:
Example IndexResult Object

Commit:
Added object examples

* BulkItemFailure Object
* Result Object

Commit:
Removed future tense and future references

* Left FUTURE as a parameter option and instances of future where it
relates to functionality, i.e., update requests can be scheduled for a
future time.
* Rewrote future tense to present tense where it was possible to do so.
* Deleted any reference to capability that might be added at a future
date.

Commit:
Cleanup

* Found more future tense in snippets
* Rewrote any (s) or (es)

Commit:
Clarified version for row object

It is often returned with a Row Object, but it’s outside the brackets
and represents sheet.version. The other idea I had was to just remove
it from this table completely. It’s not technically in the object
brackets.

Commit:
Clarified paging for List Reports

The initial work item claimed that “loadAll” doesn’t work for List
Reports, but we don’t actually use “loadAll” for this type (Reports). I
verified that you can use paging using any of the 5 options (cURL,
Java, etc.). The only confusing one was Node.js, so I added an example
of how that works.

Commit:
Clarified 4000 character limit

Commit:
Clarified modifiedSince

* Proposed using Get Sheet Version instead.
* modifiedAt only seems to be a response value.

Commit:
Clarified pagination and reports

Pagination is wonky in reports. It semi-works in Java and C#. Per Dave,
it’s not supported for Reports.

Commit:
Clarified Cell History

displayValue shows current formatting (currently in review with Rachel
Singh)

Commit:
Undid cell history change

Still parsing this one, but my initial fix didn’t really fix the
problem. I left in the link to Cell Reference.

Commit:
Clarified hyperlinks

This is related to the Get Cell History issue. I think the user just
needs some guidance that the hyperlink object is its own thing rather
than listed in cell.value or cell.displayValue.

Commit:
Simplified.

Commit:
Prep for release

seweil commented on source/snippets/addRowUpdateRow.md.erb:
```diff
@@ -1 +1 @@
-<li>One or more [*location-specifier* attributes](#row-location) (optional)</li><li>**expanded** (optional)</li><li>**format** (optional)</li><li>**cells** (optional) -- if specified, must be an array of [Cell objects](#cell-object), where each object is limited to the following attributes:</li><ul><li>columnId (required)</li><li>One of the following (required):</li><ul><li>value</li><li>formula< ...
```
Why removing the Start Date note?

kfansler replied:
This relates to APIDOC-1068. I interpreted that work item as meaning that the Predecessor flag no longer blocks you from modifying the start date. There is still a restriction on start date for a parent row, but that is called out in a note in the text.

kfansler replied:
I put the Start Date note back in based on email with Dave.

seweil commented on source/snippets/listAllOrgSheets.md.erb:
```diff
@@ -49,7 +49,7 @@ sheets = response.data
 `GET /users/sheets`
 
-Gets the list of all Sheets owned by the members of the account (organization). 
+Gets a summarized list of all Sheets owned by the members of the account (organization). 
```
I'm afraid this will be interpreted as returning a partial list. We are returning a full list - with summary details for each sheet. Same for listSentUpdateRequests.md.erb & listUpdateRequests.md.erb

kfansler replied:
Good point. I thought I had figured out an elegant solution, but I see that it is ambiguous. I rewrote 27 topics using "summarized" so I can fix all of those instances. How about this as a generic wording fix that I may have to tweak a bit in some of those cases? "Gets a list of all Sheets... " And then a 2nd sentence saying: "The list contains only summary details for each Sheet."

kfansler replied:
It looks like you only get an abbreviated object for Sheets and Workspaces, so I've corrected the doc to reflect that.

seweil commented on source/index.html.md.erb:
```diff
@@ -27,7 +27,7 @@ search: false
 <% HEADER_CT_JSON = "**Content-Type**: application/json" %>
 <% PARAMETER_ALLOW_PARTIAL_SUCCESS = "**allowPartialSuccess** (optional): when specified with a value of `true`, enables partial success for this bulk operation.  See [Partial Success](#partial-success) for more information." %>
-<% PARAMETER_MODIFIEDSINCE = "**modifiedSince** (optional): when specified with a <a href='#dates-and-times'>date and time</a> value, response only includes the objects that are modified on or after the date and time specified." %>
+<% PARAMETER_MODIFIEDSINCE = "**modifiedSince** (optional): when specified with a <a href='#dates-and-times'>date and time</a> value, response only includes the objects that are modified on or after the date and time specified. The return value is only accurate within 1-2 hours because of lag time. If you need more precise information about updates, use [Get Sheet Version](#get-sheet-version)." % ...
```
Let's refine this text. (Also this text shows up for reports as well as sheets)

kfansler replied:
How about something more like this: "If you need to keep track of frequent changes, it may be more useful to  use "Get Sheet Version."

kfansler replied:
Done.

seweil commented on source/index.html.md.erb:
```diff
@@ -325,10 +335,14 @@ Before you begin, you should already have an access token, which you used in the
 	"id": 6141831453927300,                         // This is the Id of the entire sheet
 	"name": "My first sheet",
-	"columns": [{                                   // Each Column Object connects column title to Id and defines the column details
+    "columns": [{                                   // Each Column Object connects column
```
"connectes column title to Id" --> "associates column Id with column title" ??

kfansler replied:
Done.

seweil commented on source/index.html.md.erb:
```diff
@@ -342,16 +356,22 @@ Before you begin, you should already have an access token, which you used in the
+
 					"value": "John Doe",            // Actual cell value
-					"displayValue": "John Doe"      // How the cell value is displayed in the UI
+                    "displayValue": "John Doe"      // How the cell value is displayed
```
This comment would be more helpful on the next, numeric cell. value could be 12345 and displayValue could be "12,345"

kfansler replied:
Done

seweil commented on source/index.html.md.erb:
```diff
@@ -556,11 +576,11 @@ redirect_uri<br/>(optional) | Redirect URL you [registered](#register-your-app)
 `https://app.smartsheet.com/b/authorize?response_type=code&client_id=dheu3dmkd32fhxme&scope=READ_SHEETS%20WRITE_SHEETS&state=MY_STATE`
  
-If your user has not logged in to Smartsheet, he/she will first be directed to the Smartsheet login page. 
+If your user has not logged in to Smartsheet, he/she is first directed to the Smartsheet login page. 
```
I generally prefer "they are" to "he/she is"

kfansler replied:
Done.

seweil commented on source/index.html.md.erb:
```diff
@@ -3652,8 +3674,8 @@ Date			| DATE			|
 Duration		| DURATION		| **Only for dependency-enabled project sheets**<br/>The API does not support setting a Column to this type. (This can only be done through the Smartsheet Web app when configuring a project sheet.)
-Predecessor		| PREDECESSOR	| **Only for dependency-enabled project sheets**<br/>The API does not support setting a Column to this type, or updating data in a column of this type. (This can only be done through the Smartsheet Web app when configuring a project sheet.) 
-Date/Time 		| ABSTRACT_DATETIME | Represents a project sheet's Start and End dates.<br/>**Only for dependency-enabled project sheets**<br/>The API does not support setting a Column to this type. (This can only be done through the Smartsheet Web app when configuring a project sheet.) Additionally, the API does not support updating data in the End Date column under any circumstance, and does not su ...
+Predecessor		| PREDECESSOR	| **Only for dependency-enabled project sheets** 
```
Refer to predecessor object?

kfansler replied:
Agreed. Done.

Commit:
Fixes based on review comments.

seweil commented on source/index.html.md.erb:
```diff
@@ -8064,18 +8062,15 @@ reports = response.data
 `GET /reports`
 
-Gets the list of all Reports that the User has access to, in alphabetical order, by name.
+Gets a summarized list of all Reports that the User has access to in alphabetical order by name. This index API does not support pagination.
```
Experimentally, it *does*

kfansler replied:
This one is long and convoluted.  I included my UPDATE in the description for this one: APIDOC-1060. I suspect it isn't supported because it is inconsistent?

seweil commented on source/index.html.md.erb:
```diff
@@ -14538,7 +14524,7 @@ emails = response.data
 `GET /users/{userId}/alternateemails`
 
-Gets a list of the Alternate Emails for the specified User. 
+Gets a summarized list of the Alternate Emails for the specified User. 
```
Not full objects?

kfansler replied:
Hmm. There are only 3 things in an object, but I can't test this. I just get a data[] back when I try to test it on myself. Do you have a sample user I could use?

kfansler replied:
The docs show the full object. I have a query in to Dave to see if he can actually test it to verify.

kfansler replied:
Tested. You do get the full object back. I'll rewrite.

kfansler replied:
Done.

seweil commented on source/index.html.md.erb:
````diff
@@ -15535,6 +15521,16 @@ For details about working with sheets in a Workspace, see [Sheets](#sheets).</br
+
+```json
+{
+    "prefix": "APIDOC-",
````
How about "Item-" or "Order-" ?

kfansler replied:
Done

Commit:
Made more generic

Commit:
Finished incorporating review comments

Commit:
Streamlined List Discussions response

Removed the Attachments array as it muddies the point of the sample.

kfansler committed:
Put Start Date restriction note back in

Dave filed a bug to sort this out, but for now, the UI response is too
confusing for it to be called a “finished” thing. Putting this part of
the doc back how it was.

kfansler committed:
Incorporated review changes

kfansler committed:
Predecessor solution
